chia/wallet/puzzles/dao_proposal_validator.clsp

```diff
@@ -42,70 +36,73 @@
   )
 
   (defun-inline calculate_proposal_puzzle (
-      PROPOSAL_MOD_HASH
-      PROPOSAL_SINGLETON_STRUCT
-      PROPOSAL_TIMER_MOD_HASH
-      CAT_MOD_HASH
-      DAO_FINISHED_STATE_MOD_HASH
-      TREASURY_MOD_HASH
-      LOCKUP_MOD_HASH
-      CAT_TAIL_HASH
-      TREASURY_ID
+      PROPOSAL_SELF_HASH
+      proposal_singleton_id
       proposal_yes_votes
       proposal_total_votes
       proposal_innerpuz_hash
     )
-    (puzzle-hash-of-curried-function PROPOSAL_MOD_HASH
-      (sha256 ONE proposal_innerpuz_hash)
+    (puzzle-hash-of-curried-function PROPOSAL_SELF_HASH
       (sha256 ONE proposal_total_votes)
       (sha256 ONE proposal_yes_votes)
-      (sha256 ONE TREASURY_ID)
-      (sha256 ONE CAT_TAIL_HASH)
-      (sha256 ONE LOCKUP_MOD_HASH)
-      (sha256 ONE TREASURY_MOD_HASH)
-      (sha256 ONE DAO_FINISHED_STATE_MOD_HASH)
-      (sha256 ONE CAT_MOD_HASH)
-      (sha256 ONE PROPOSAL_TIMER_MOD_HASH)
-      (sha256 ONE PROPOSAL_MOD_HASH)
-      (sha256tree PROPOSAL_SINGLETON_STRUCT)
+      (sha256 ONE proposal_innerpuz_hash)
+      (sha256 ONE proposal_singleton_id)
+      (sha256 ONE PROPOSAL_SELF_HASH)
     )
   )
 
-  (assert
+  (if
     ; (= (sha256tree my_solution) announcement_args) - quex suggested this. We don't need to check it now. Can be used for future functionality.
     (> (+ coin_amount ONE) PROPOSAL_MINIMUM_AMOUNT)  ; >=
-    (> total_votes Attendance_Required)  ; TODO: we might want to change this to storing total cats and calculating like with yes votes
-    (> yes_votes (calculate_win_percentage total_votes Pass_Margin))
-    (=
-      announcement_source
-      (calculate_coin_id
-        coin_parent
-        (calculate_full_puzzle_hash
-          (c (f SINGLETON_STRUCT) (c proposal_id (r (r SINGLETON_STRUCT))))
-          (calculate_proposal_puzzle
-            PROPOSAL_MOD_HASH
-            (c (f SINGLETON_STRUCT) (c proposal_id (r (r SINGLETON_STRUCT))))
-            PROPOSAL_TIMER_MOD_HASH
-            CAT_MOD_HASH
-            DAO_FINISHED_STATE_MOD_HASH
-            TREASURY_MOD_HASH
-            LOCKUP_MOD_HASH
-            CAT_TAIL_HASH
-            (f (r SINGLETON_STRUCT))
+    (if
```

Comment:
how come we use nested if's instead of an assert now?

Reply:
for the sake of adding debug values to the (x) will fix